### PR TITLE
test: reorganize subgraph test suite into composable domain specs

### PR DESCRIPTION
**Pull request by DrJKL**
## Summary

Reorganize the subgraph test suite so browser tests are thin representative user journeys while lower-level Vitest suites own combinatorics, migration edge cases, and data-shape semantics.

## Changes

- **What**: Migrate 17 flat subgraph browser specs into 10 domain-organized specs under `browser_tests/tests/subgraph/`, move redundant semantic coverage down to 8 Vitest owner suites, delete all legacy flat files
- **Browser specs** (54 tests): `subgraphSlots`, `subgraphPromotion`, `subgraphPromotionDom`, `subgraphSerialization`, `subgraphNavigation`, `subgraphNested`, `subgraphLifecycle`, `subgraphCrud`, `subgraphSearch`, `subgraphOperations`
- **Vitest owners** (230 tests): `SubgraphNode.test.ts` (rename/label propagation), `subgraphNodePromotion.test.ts`, `promotedWidgetView.test.ts`, `SubgraphSerialization.test.ts` (duplicate-ID remap), `SubgraphWidgetPromotion.test.ts` (legacy hydration), `subgraphNavigationStore*.test.ts` (viewport cache, workflow-switch), `subgraphStore.test.ts` (search aliases, description)
- **Net effect**: browser suite shrinks from ~96 scattered tests to 54 focused journeys

## Review Focus

- Coverage ownership split: each browser test has a unique UI-only failure mode; semantic coverage lives in Vitest
- `subgraphPromotionDom.spec.ts` forces LiteGraph mode and uses `canvas.openSubgraph()` instead of `navigateIntoSubgraph()` to avoid a wrapper-specific DOM overlay duplication issue — entry-affordance coverage lives in `subgraphNavigation.spec.ts`
- No product code changes — test-only migration

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-10759-test-reorganize-subgraph-test-suite-into-composable-domain-specs-3336d73d365081b0a56bcbf809b1f584) by [Unito](https://www.unito.io)

### CODE REVIEW

File: browser_tests/tests/subgraph/subgraphCrud.spec.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  test.describe('Subgraph Creation and Deletion', () => {
+    test('Can create subgraph from selected nodes', async ({ comfyPage }) => {
+      await comfyPage.workflow.loadWorkflow('default')
+
+      await comfyPage.keyboard.selectAll()
+      await comfyPage.nextFrame()
+
+      const node = await comfyPage.nodeOps.getNodeRefById('5')
+      await node.convertToSubgraph()
+
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply browser_tests/tests/subgraph/subgraphCrud.spec.ts:76]: Declining — node ID `'5'` is a fixed value in the committed `default.json` fixture, not a runtime-generated ID. The `default` workflow is an established pattern used across dozens of tests in this codebase. Creating a dedicated fixture just for this test is over-engineering for a reorganization PR and introduces inconsistency with the rest of the test suite.

If title-based lookups across all subgraph tests become desirable, that's a separate readability improvement PR.

File: browser_tests/tests/subgraph/subgraphCrud.spec.ts [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
+    await comfyPage.page.mouse.move(subgraphPos.x + 16, subgraphPos.y + 16)
+    await comfyPage.page.keyboard.down('Alt')
+    try {
+      await comfyPage.page.mouse.down()
+      await comfyPage.nextFrame()
+
+      await comfyPage.page.mouse.move(subgraphPos.x + 64, subgraphPos.y + 64)
+      await comfyPage.page.mouse.up()
+    } finally {
+      await comfyPage.page.keyboard.up('Alt')
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply browser_tests/tests/subgraph/subgraphCrud.spec.ts:28]: Thanks — agreed the try-finally pattern is good here (it's already implemented). Extracting a suite-wide `withModifierKey` helper is out of scope for this test reorganization PR. Many existing `keyboard.down` calls involve multi-key combos that wouldn't cleanly fit a single wrapper. If the pattern proves painful, it can be a follow-up.

File: browser_tests/tests/subgraph/subgraphNavigation.spec.ts [hunk 3]
```diff
@@ -62,18 +47,12 @@ test.describe('Subgraph Navigation', { tag: ['@slow', '@subgraph'] }, () => {
...
       await subgraphNode.navigateIntoSubgraph()
 
-      await comfyPage.page.waitForSelector(SELECTORS.breadcrumb, {
-        state: 'visible',
-        timeout: 20000
-      })
-
-      const breadcrumb = comfyPage.page.locator(SELECTORS.breadcrumb)
+      const breadcrumb = comfyPage.page.getByTestId(TestIds.breadcrumb.subgraph)
+      await expect(breadcrumb).toBeVisible({ timeout: 20_000 })
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply browser_tests/tests/subgraph/subgraphNavigation.spec.ts:53]: This is a retrying assertion — Playwright resolves it as soon as the breadcrumb appears; 20s is just the CI safety net. The subgraph entry pipeline (graph setup → widget promotion → DOM render → breadcrumb mount) is legitimately heavy, and this suite is tagged `@slow`. The timeout doesn't mean it *takes* 20 seconds, just that it's *allowed* to. Reducing it risks CI flakiness without any behavioral improvement.

File: browser_tests/tests/subgraph/subgraphSlots.spec.ts [hunk 4]
```diff
@@ -452,131 +318,170 @@ test.describe('Subgraph Slots', { tag: ['@slow', '@subgraph'] }, () => {
...
+  test.describe('Subgraph promoted widget-input slot position', () => {
+    test('Promoted text widget slot is positioned at widget row, not header', async ({
       comfyPage
     }) => {
       await comfyPage.workflow.loadWorkflow(
-        'subgraphs/subgraph-compressed-target-slot'
+        'subgraphs/subgraph-with-promoted-text-widget'
+      )
+      // Two frames needed: first renders slot changes, second stabilizes layout
+      await comfyPage.nextFrame()
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply browser_tests/tests/subgraph/subgraphSlots.spec.ts:472]: The existing comment already explains *why* at the right level of abstraction. Expanding to detail what specifically happens in each frame would couple the comment to internal rendering implementation details (LiteGraph layout passes, Vue reactivity flush order) that are speculative from the test's perspective and subject to change. Keeping it as-is.

File: src/lib/litegraph/src/subgraph/SubgraphNode.test.ts [hunk 5]
```diff
@@ -985,21 +986,43 @@ describe('SubgraphNode label propagation', () => {
...
     subgraph.renameInput(subgraph.inputs[0], 'Steps Count')
 
-    expect(subgraphNode.inputs[0].label).toBe('Steps Count')
-    expect(subgraphNode.inputs[0].name).toBe('steps')
+    expect(promotedInput.label).toBe('Steps Count')
+    expect(promotedInput.name).toBe('steps')
+    expect(promotedInput.widget?.name).toBe(originalWidgetName)
+    expect(promotedInput._widget?.label).toBe('Steps Count')
+    expect(subgraphNode.widgets?.[0].label).toBe('Steps Count')
+    expect(labelChangedSpy).toHaveBeenCalledWith('node:slot-label:changed', {
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply src/lib/litegraph/src/subgraph/SubgraphNode.test.ts:1020]: Thanks for the positive note! Agreed that event verification is valuable for ensuring downstream consumers stay notified. If specific coverage gaps in other state mutations are identified, they can be tracked as a separate issue — out of scope for this reorg PR.

File: browser_tests/tests/subgraph/subgraphNavigation.spec.ts [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+      nx + nw > visLeft &&
+      nx < visRight &&
+      ny + nh > visTop &&
+      ny < visBottom
+    )
+      return true
+  }
+
+  return false
+}
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply browser_tests/tests/subgraph/subgraphNavigation.spec.ts:32]: Declining -- partial overlap is the correct semantic here. The function answers 'can I see at least part of a node on screen?' A node half-clipped at the viewport edge is still visible to the user. Requiring full containment would cause false negatives for large nodes or nodes near edges, making the test flaky for no behavioral benefit. The function name hasVisibleNodeInViewport already describes the intended behavior accurately.

File: browser_tests/tests/subgraph/subgraphNested.spec.ts [hunk 7]
```diff
@@ -4,177 +4,44 @@ import { comfyPageFixture as test, comfyExpect } from '../../fixtures/ComfyPage'
...
-          expect(Number(interiorNodeId)).toBeGreaterThan(0)
-          expect(widgetName).toBeTruthy()
-          expect(
-            resolved,
-            `Widget "${widgetName}" (interior node ${interiorNodeId}) on host ${hostNodeId} should resolve`
-          ).toBe(true)
-        }
+        const response = await responsePromise
+        expect(warnings).toEqual([])
+        expect(response.ok()).toBe(true)
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply browser_tests/tests/subgraph/subgraphNested.spec.ts:28]: Declining -- this test validates subgraph structural integrity (nested subgraphs with duplicate IDs load and queue without resolution failures), not API contract compliance. response.ok() is the right semantic: the server accepted the prompt without error. Pinning to 200 couples the test to an endpoint implementation detail. Asserting response body structure turns this into an API contract test, which belongs elsewhere. The current assertion was already an intentional improvement from status !== 400 per earlier review feedback.

**Review by christian-byrne — APPROVED** [review]
